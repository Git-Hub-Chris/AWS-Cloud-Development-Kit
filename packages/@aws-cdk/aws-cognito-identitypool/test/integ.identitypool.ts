import {
  UserPool,
} from '@aws-cdk/aws-cognito';
import {
  Effect,
  PolicyStatement,
} from '@aws-cdk/aws-iam';
import {
  App,
  Stack,
} from '@aws-cdk/core';
import {
  IdentityPool,
} from '../lib/identitypool';
import {
  UserPoolAuthenticationProvider,
} from '../lib/identitypool-user-pool-authentication-provider';

const app = new App();
const stack = new Stack(app, 'integ-identitypool');

const userPool = new UserPool(stack, 'Pool');
const otherPool = new UserPool(stack, 'OtherPool');

const idPool = new IdentityPool(stack, 'identitypool', {
  authenticationProviders: {
    userPool: new UserPoolAuthenticationProvider({ userPool }),
    amazon: { appId: 'amzn1.application.12312k3j234j13rjiwuenf' },
    google: { clientId: '12345678012.apps.googleusercontent.com' },
  },
  allowClassicFlow: true,
  identityPoolName: 'my-id-pool',
});
idPool.authenticatedRole.addToPrincipalPolicy(new PolicyStatement({
  effect: Effect.ALLOW,
  actions: ['dynamodb:*'],
  resources: ['*'],
}));
idPool.unauthenticatedRole.addToPrincipalPolicy(new PolicyStatement({
  effect: Effect.ALLOW,
  actions: ['dynamodb:Get*'],
  resources: ['*'],
}));
idPool.addUserPoolAuthentication(new UserPoolAuthenticationProvider({ userPool: otherPool }));
app.synth();