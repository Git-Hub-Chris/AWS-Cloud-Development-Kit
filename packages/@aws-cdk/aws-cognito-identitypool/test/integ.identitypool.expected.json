{
  "Resources": {
    "PoolD3F588B8": {
      "Type": "AWS::Cognito::UserPool",
      "Properties": {
        "AccountRecoverySetting": {
          "RecoveryMechanisms": [
            {
              "Name": "verified_phone_number",
              "Priority": 1
            },
            {
              "Name": "verified_email",
              "Priority": 2
            }
          ]
        },
        "AdminCreateUserConfig": {
          "AllowAdminCreateUserOnly": true
        },
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new account",
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "VerificationMessageTemplate": {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new account",
          "SmsMessage": "The verification code to your new account is {####}"
        }
      },
      "UpdateReplacePolicy": "Retain",
      "DeletionPolicy": "Retain"
    },
    "PoolUserPoolClientForundefinedBF6BDE57": {
      "Type": "AWS::Cognito::UserPoolClient",
      "Properties": {
        "UserPoolId": {
          "Ref": "PoolD3F588B8"
        },
        "AllowedOAuthFlows": [
          "implicit",
          "code"
        ],
        "AllowedOAuthFlowsUserPoolClient": true,
        "AllowedOAuthScopes": [
          "profile",
          "phone",
          "email",
          "openid",
          "aws.cognito.signin.user.admin"
        ],
        "CallbackURLs": [
          "https://example.com"
        ],
        "SupportedIdentityProviders": [
          "COGNITO"
        ]
      }
    },
    "OtherPool7DA7F2F7": {
      "Type": "AWS::Cognito::UserPool",
      "Properties": {
        "AccountRecoverySetting": {
          "RecoveryMechanisms": [
            {
              "Name": "verified_phone_number",
              "Priority": 1
            },
            {
              "Name": "verified_email",
              "Priority": 2
            }
          ]
        },
        "AdminCreateUserConfig": {
          "AllowAdminCreateUserOnly": true
        },
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new account",
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "VerificationMessageTemplate": {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new account",
          "SmsMessage": "The verification code to your new account is {####}"
        }
      },
      "UpdateReplacePolicy": "Retain",
      "DeletionPolicy": "Retain"
    },
    "OtherPoolUserPoolClientForundefined1B97829F": {
      "Type": "AWS::Cognito::UserPoolClient",
      "Properties": {
        "UserPoolId": {
          "Ref": "OtherPool7DA7F2F7"
        },
        "AllowedOAuthFlows": [
          "implicit",
          "code"
        ],
        "AllowedOAuthFlowsUserPoolClient": true,
        "AllowedOAuthScopes": [
          "profile",
          "phone",
          "email",
          "openid",
          "aws.cognito.signin.user.admin"
        ],
        "CallbackURLs": [
          "https://example.com"
        ],
        "SupportedIdentityProviders": [
          "COGNITO"
        ]
      }
    },
    "identitypool51BC8967": {
      "Type": "AWS::Cognito::IdentityPool",
      "Properties": {
        "AllowUnauthenticatedIdentities": false,
        "AllowClassicFlow": true,
        "CognitoIdentityProviders": [],
        "IdentityPoolName": "my-id-pool",
        "SupportedLoginProviders": {
          "www.amazon.com": "amzn1.application.12312k3j234j13rjiwuenf",
          "accounts.google.com": "12345678012.apps.googleusercontent.com"
        }
      }
    },
    "identitypoolidentitypoolAuthenticatedRole8591A38F": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRoleWithWebIdentity",
              "Condition": {
                "StringEquals": {
                  "cognito-identity.amazonaws.com:aud": {
                    "Ref": "identitypool51BC8967"
                  }
                },
                "ForAnyValue:StringLike": {
                  "cognito-identity.amazonaws.com:amr": "authenticated"
                }
              },
              "Effect": "Allow",
              "Principal": {
                "Federated": "cognito-identity.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "Description": "Default Authenticated Role for Identity Pool my-id-pool",
        "RoleName": "identitypoolAuthenticatedRole"
      }
    },
    "identitypoolidentitypoolAuthenticatedRoleDefaultPolicy8660132D": {
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "dynamodb:*",
              "Effect": "Allow",
              "Resource": "*"
            }
          ],
          "Version": "2012-10-17"
        },
        "PolicyName": "identitypoolidentitypoolAuthenticatedRoleDefaultPolicy8660132D",
        "Roles": [
          {
            "Ref": "identitypoolidentitypoolAuthenticatedRole8591A38F"
          }
        ]
      }
    },
    "identitypoolidentitypoolUnauthenticatedRoleCEE08F6C": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRoleWithWebIdentity",
              "Condition": {
                "StringEquals": {
                  "cognito-identity.amazonaws.com:aud": {
                    "Ref": "identitypool51BC8967"
                  }
                },
                "ForAnyValue:StringLike": {
                  "cognito-identity.amazonaws.com:amr": "unauthenticated"
                }
              },
              "Effect": "Allow",
              "Principal": {
                "Federated": "cognito-identity.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "Description": "Default Unauthenticated Role for Identity Pool my-id-pool",
        "RoleName": "identitypoolUnauthenticatedRole"
      }
    },
    "identitypoolidentitypoolUnauthenticatedRoleDefaultPolicyA8F8549F": {
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "dynamodb:Get*",
              "Effect": "Allow",
              "Resource": "*"
            }
          ],
          "Version": "2012-10-17"
        },
        "PolicyName": "identitypoolidentitypoolUnauthenticatedRoleDefaultPolicyA8F8549F",
        "Roles": [
          {
            "Ref": "identitypoolidentitypoolUnauthenticatedRoleCEE08F6C"
          }
        ]
      }
    },
    "identitypoolidentitypoolDefaultRoleAttachmentB154BD67": {
      "Type": "AWS::Cognito::IdentityPoolRoleAttachment",
      "Properties": {
        "IdentityPoolId": {
          "Ref": "identitypool51BC8967"
        },
        "Roles": {
          "authenticated": {
            "Fn::GetAtt": [
              "identitypoolidentitypoolAuthenticatedRole8591A38F",
              "Arn"
            ]
          },
          "unauthenticated": {
            "Fn::GetAtt": [
              "identitypoolidentitypoolUnauthenticatedRoleCEE08F6C",
              "Arn"
            ]
          }
        }
      },
      "DependsOn": [
        "identitypool51BC8967",
        "identitypoolidentitypoolAuthenticatedRoleDefaultPolicy8660132D",
        "identitypoolidentitypoolAuthenticatedRole8591A38F",
        "identitypoolidentitypoolUnauthenticatedRoleDefaultPolicyA8F8549F",
        "identitypoolidentitypoolUnauthenticatedRoleCEE08F6C"
      ]
    }
  }
}